### ISSUE TITLE
compositor crash with async reprojection

### ISSUE THREAD

**Issue by ChristophHaag** [ValveSoftware/SteamVR-for-Linux#184]
#### Your system information
* Steam client version (build number or date): Mar 8 2019 at 19:34:27
* Distribution (e.g. Ubuntu): Archlinux
* Graphics driver version (run nvidia-settings): mesa git ad259482610
* Gist for SteamVR System Information: 
[SteamVR-2019-03-09-_11_21_48.txt](https://github.com/ValveSoftware/SteamVR-for-Linux/files/2948895/SteamVR-2019-03-09-_11_21_48.txt)
* Opted into Steam client beta?: [Yes/No] Yes
* Opted into SteamVR beta?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:
With my Vega 64, vrcompositor often crashes shortly after async reprojection kicks in with this backtrace
```
Core was generated by `/home/chris/.local/share/Steam/steamapps/common/SteamVR/bin/linux64/vrcomposito'.
Program terminated with signal SIGABRT, Aborted.
#0  0x00007f926f175d7f in raise () from /usr/lib/libc.so.6
[Current thread is 1 (Thread 0x7f926c629700 (LWP 17129))]
(gdb) bt
#0  0x00007f926f175d7f in raise () from /usr/lib/libc.so.6
#1  0x00007f926f160672 in abort () from /usr/lib/libc.so.6
#2  0x0000000000544ebb in CThreadWatchdog::EvaluateWatchdogs() ()
#3  0x0000000000544f9d in std::thread::_Impl<std::_Bind_simple<CThreadWatchdog::InitWatchdogs()::{lambda()#1} ()> >::_M_run() ()
#4  0x00007f926f3e1424 in std::execute_native_thread_routine_compat (__p=<optimized out>) at /build/gcc/src/gcc/libstdc++-v3/src/c++11/thread.cc:94
#5  0x00007f926f30ba9d in start_thread () from /usr/lib/libpthread.so.0
#6  0x00007f926f239b23 in clone () from /usr/lib/libc.so.6
```
[And here is the backtrace of all threads](https://github.com/ValveSoftware/SteamVR-for-Linux/files/2948899/thread_apply_all_bt.txt).

When keeping an application's gpu usage low enough that reprojection is not necessary, the crash does not seem to happen. It seems to happen shortly after reprojection kicks in.

This is a frame timing graph while teleporting around in steamvr home. When vr compositor crashes, it conveniently stops updating and stays like this:
![Screenshot_20190309_232531](https://user-images.githubusercontent.com/781076/54078093-acb2d300-42c2-11e9-9503-f2b6d37af4ed.png)

#### Steps for reproducing this issue:
1. Run a GPU heavy application like SteamVR Home.

**Comment by lostgoat** [comment]
As a workaround you can close the mirror window.

A proper fix will be part of the upcoming beta build.

**Comment by lostgoat** [comment]
Should be fixed in 1.3.8
